src/game/sound_samples.c

```diff
@@ -84,6 +86,23 @@ int init_buffer_and_device(Sound_samples *sound_samples,
         }
     }
     
+    /* Allocating active audio buffer location*/
+    //TODO: Allocate one huge active audio buffer with length of the maximum of all audio buffer, instead of one active buffer for each audio
+    sound_samples->active_audio_buf_array = PUSH_LT(sound_samples->lt, nth_calloc(sound_samples->samples_count, sizeof(uint8_t*)), free);
+    if (sound_samples->active_audio_buf_array == NULL) {
+      log_fail("Failed to allocate memory for active audio buffer pointer array\n");
+      return -1;
```

Comment:
Use RETURN_LT here.

Reply:
Why it needs a `RETURN_LT` here while the other `return` in this function uses a raw `return -1;`?

Reply:
@PaprikaX33 oh! I actually misunderstood this code! We don't need `RETURN_LT` because the `lt` will be destroyed by the caller of `init_buffer_and_device` in case of an error (`create_sound_samples`). Forget the RETURN_LT remarks! Thank you for asking for clarification!